lex.go

```diff
@@ -189,14 +191,36 @@ func lexBeforeKey(l *lexer) stateFn {
 		return nil
 
 	case isEOL(r):
-		l.ignore()
-		return lexBeforeKey
+		if l.preserveFormatting {
```

Comment:
Please make these two separate switch cases (also for the other ones):

```go
case isEOL(r) && l.keepWS:
    l.appendRune(r)
    l.backup()
    return lexComment

case isEOL(r):
    l.ignore()
    return lexBeforeKey
```

Reply:
Done.